modules/core/arrow-syntax/src/main/kotlin/arrow/syntax/collections/tuple.generator.kts

```diff
@@ -0,0 +1,48 @@
+import java.io.File
+
+val notice = """
+/*
+
+  AUTOGENERATED FILE. DON'T EDIT DIRECTLY
+
+  This file was generated by tuple.generator.kts
+
+*/
+
+"""
+
+val package_ = "arrow.syntax.collections"
+
+val maxTuple = 21
+
+val availableLetters = ('a'..'z').toList().map { it.toString() }
+
+val fileContent = StringBuilder()
+
+val imports = (2..maxTuple + 1).joinToString(separator = "\n") { "import arrow.core.Tuple$it" }
+
+fileContent.append(notice)
+fileContent.append("package $package_")
+fileContent.append("\n\n")
+fileContent.append(imports)
+fileContent.append("\n\n")
+
+for (i in 2..maxTuple) {
+
+  val letters = availableLetters.take(i + 1)
+
+  val diamond1 = "<${letters.joinToString { it.toUpperCase() }}>"
+  val diamond2 = "<${letters.dropLast(1).joinToString { it.toUpperCase() }}>"
+
+  val newLetter = availableLetters[i]
+
+  val constructor = (letters.dropLast(1).map { "this.$it" } + newLetter).joinToString()
+
+  fileContent.append(
+          "operator fun $diamond1 Tuple$i$diamond2.plus($newLetter: ${newLetter.toUpperCase()}): Tuple${i+1}$diamond1 = Tuple${i + 1}($constructor)"
+  )
+  fileContent.append("\n")
+
+}
+
+File("tuple.kt").writeText(fileContent.toString())
```

Comment:
This is too useful to make it throwaway code, we can modify it for other functions. It cannot be shipped with the library tho, can you please move it to...

...to where?

Can you create a folder in the root of the project called scripts and drop it there? Will it even work? How did you invoke it?

Reply:
Those scripts can live in the annotation processor which is not a runtime dependency